consumer/nf_discovery.go

```diff
@@ -33,6 +44,20 @@ func SendSearchNFInstances(nrfUri string, targetNfType, requestNfType models.NfT
 			err = fmt.Errorf("SearchNFInstances' response body cannot close: %+w", bodyCloseErr)
 		}
 	}()
+
+	// subscribe only if caching is enabled?
+	if amf_context.AMF_Self().EnableNrfCaching {
+		amfSelf := amf_context.AMF_Self()
+		for _, nfProfile := range result.NfInstances {
+			nrfSubscriptionData := models.NrfSubscriptionData{
+				NfStatusNotificationUri: fmt.Sprintf("%s/namf-callback/v1/nf-status-notify", amfSelf.GetIPv4Uri()),
+				SubscrCond:              &models.NfInstanceIdCond{NfInstanceId: nfProfile.NfInstanceId},
+				ReqNfType:               requestNfType,
+			}
+			SendCreateSubscription(nrfUri, nrfSubscriptionData)
```

Comment:
Handled the error.
SPEC suggests used of NFInstanceIdCond, NfInstanceIdListCond, NfTypeCond etc. to specify the subscription condition. 
It would be better to use NfInstanceIdListCond, but that is not currently supported in the openapi models.
Supported ones are nf_group_cond, nf_instance_id_cond and nf_type_cond. Most appropriate seemed to be nf_instance_id_cond. This is why subscription is sent individually for each nf instance in the nf profile list.